### PR TITLE
Inc map

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: core/arch/arm/include/mm/mobj.h [hunk 1]
```diff
@@ -74,6 +76,40 @@ static inline bool mobj_matches(struct mobj *mobj, enum buf_is_attr attr)
...
+	return TEE_ERROR_NOT_SUPPORTED;
+}
+
+/**
+ * mobj_dec_map() - decrease map count
+ * @mobj:	pointer to a registered shared memory MOBJ
+ *
+ * Decreases the map count and also unmaps the MOBJ if the map count
+ * reaches 0.  Each call to mobj_reg_shm_inc_map() is supposed to be
+ * matched by a call to mobj_reg_shm_dec_map().
```
Comment (jforissier) [inline core/arch/arm/include/mm/mobj.h:102]: mobj_inc_map(), mobj_dec_map()

**Comment by jforissier** [comment]
Reviewing commit "core: mobj: assign dummy {inc,dec}_map() to common mobj types".

Why is this needed? We could just leave the pointers NULL and that would be taken care of by `mobj_inc_map()` and `mobj_dec_map()`, no?

**Comment by jenswi-linaro** [comment]
Dropping "core: mobj: assign dummy {inc,dec}_map() to common mobj types" and dealing with that in the inline functions instead.

Tags applied.

@sahilnxp, needed something like this and then I noticed that "core: always increase mappings for pta memrefs" seemed like a good idea.

**Comment by jforissier** [comment]
OK, cool.

**Comment by jforissier** [comment]
Looks like the sentence "If struct mobj_ops for a particular mobj doesn't have and implementation of
inc_map() or dec_map() TEE_ERROR_NOT_SUPPORTED is returned instead." should be removed from the first commit description since you're returning success.

**Comment by jenswi-linaro** [comment]
> Looks like the sentence "If struct mobj_ops for a particular mobj doesn't have and implementation of
> inc_map() or dec_map() TEE_ERROR_NOT_SUPPORTED is returned instead." should be removed from the first commit description since you're returning success.

Sorry, I missed that. I've updated it to "..TEE_SUCCESS is returned..." instead.

**Pull request merged**